pynucastro/rates/rate.py

```diff
@@ -920,11 +921,27 @@ def eval(self, T, rhoY=None):
             r = data[inde][5]
 
         else:
-            tf = Tfactors(T)
-            r = 0.0
-            for s in self.sets:
-                f = s.f()
-                r += f(tf)
+
+            if (check_T_min and T < 1.0e8):
+
+                T_min = minimize(self._find_rate, 1.0e7, method="Nelder-Mead", tol=1.0e5).x[0]
+                assert T > T_min, f"T must be greater than {T_min} K to get reasonable rates"
+
+            r = self._find_rate(T)
+
+        return r
+
+    def _find_rate(self, T):
```

Comment:
why do you need this function instead of just using r.eval()?

Reply:
do I not need a function in order to minimize the rates? Or do you mean its better to create it inside eval() so there is no extra function that do the same thing.

Reply:
I guess O was thinking that you could do `minimize(self.eval, ...)`

Reply:
but aren't we checking T_min inside eval()?

Reply:
oh, you're right.